File: app/src/main/java/com/duzhaokun123/bilibilihd/ui/JumpActivity.java
package com.duzhaokun123.bilibilihd.ui;

import android.app.Activity;
import android.content.ComponentName;
import android.content.Intent;
import android.util.Log;

import com.duzhaokun123.bilibilihd.R;
import com.duzhaokun123.bilibilihd.databinding.ActivityJumpBinding;
import com.duzhaokun123.bilibilihd.bases.BaseActivity;
import com.duzhaokun123.bilibilihd.utils.Logcat;
import com.duzhaokun123.bilibilihd.utils.TipUtil;

public class JumpActivity extends BaseActivity<ActivityJumpBinding> {

    private Intent intent;

    @Override
    protected int initConfig() {
        return FIX_LAYOUT;
    }

    @Override
    public int initLayout() {
        return R.layout.activity_jump;
    }

    @Override
    public void initView() {
        intent = new Intent();

        baseBind.btnAdd.setOnClickListener(v -> {
            try {
                switch (baseBind.rgChoose.getCheckedRadioButtonId()) {
                    case R.id.rb_int:
                        intent.putExtra(baseBind.etKey.getText().toString(), Integer.parseInt(baseBind.etValue.getText().toString()));
                        break;
                    case R.id.rb_long:
                        intent.putExtra(baseBind.etKey.getText().toString(), Long.parseLong(baseBind.etValue.getText().toString()));
                        break;
                    case R.id.rb_double:
                        intent.putExtra(baseBind.etKey.getText().toString(), Double.parseDouble(baseBind.etValue.getText().toString()));
                        break;
                    case R.id.rb_float:
                        intent.putExtra(baseBind.etKey.getText().toString(), Float.parseFloat(baseBind.etValue.getText().toString()));
                        break;
                    case R.id.rb_boolean:
                        intent.putExtra(baseBind.etKey.getText().toString(), Boolean.parseBoolean(baseBind.etValue.getText().toString()));
                        break;
                    case R.id.rb_string:
                        intent.putExtra(baseBind.etKey.getText().toString(), baseBind.etValue.getText().toString());
                        break;
                }
                Log.d("JumpActivity", baseBind.etKey.getText().toString() + ": " + baseBind.etValue.getText().toString());
            } catch (Exception e) {
                e.printStackTrace();
                TipUtil.showToast(e.getMessage());
            }
        });

        baseBind.btnStart.setOnClickListener(v -> {
            Log.d("JumpActivity", "start" + ": " + baseBind.etComponent.getText().toString());
            try {
                Class<?> clazz = Class.forName(baseBind.etComponent.getText().toString());
                if (Activity.class.isAssignableFrom(clazz)) {
                    intent.setComponent(new ComponentName(JumpActivity.this, clazz));
                    startActivity(intent);
                } else {
                    TipUtil.showToast(clazz.getName() + " is not a Activity");
                }
            } catch (ClassNotFoundException e) {
                e.printStackTrace();
                TipUtil.showToast("class not found " + e.getMessage());
            }
        });
        baseBind.btnSaveLog.setOnClickListener(v -> Logcat.saveLog(this));
    }

    @Override
    public void initData() {

    }
}

File: app/src/main/java/com/duzhaokun123/bilibilihd/ui/JumpActivity.kt
package com.duzhaokun123.bilibilihd.ui

import android.app.Activity
import android.content.ComponentName
import android.content.Intent
import android.util.Log
import androidx.core.view.updatePadding
import com.duzhaokun123.bilibilihd.R
import com.duzhaokun123.bilibilihd.bases.BaseActivity2
import com.duzhaokun123.bilibilihd.databinding.ActivityJumpBinding
import com.duzhaokun123.bilibilihd.utils.TipUtil

class JumpActivity : BaseActivity2<ActivityJumpBinding>() {
    override fun initConfig() = setOf<Config>()
    override fun initLayout() = R.layout.activity_jump

    override fun initView() {
        baseBind.nsv.setOnScrollChangeListener(AutoSetActionBarUpListener())
        val intent = Intent()
        baseBind.btnAdd.setOnClickListener {
            try {
                when (baseBind.rgChoose.checkedRadioButtonId) {
                    R.id.rb_int -> intent.putExtra(baseBind.etKey.text.toString(), baseBind.etValue.text.toString().toInt())
                    R.id.rb_long -> intent.putExtra(baseBind.etKey.text.toString(), baseBind.etValue.text.toString().toLong())
                    R.id.rb_double -> intent.putExtra(baseBind.etKey.text.toString(), baseBind.etValue.text.toString().toDouble())
                    R.id.rb_float -> intent.putExtra(baseBind.etKey.text.toString(), baseBind.etValue.text.toString().toFloat())
                    R.id.rb_boolean -> intent.putExtra(baseBind.etKey.text.toString(), baseBind.etValue.text.toString().toBoolean())
                    R.id.rb_string -> intent.putExtra(baseBind.etKey.text.toString(), baseBind.etValue.text.toString())
                }
                Log.d(className, "${baseBind.etKey.text}: ${baseBind.etValue.text}")
            } catch (e: Exception) {
                e.printStackTrace()
                TipUtil.showToast(e.message)
            }
        }
        baseBind.btnStart.setOnClickListener {
            Log.d(className, "start: ${baseBind.etComponent.text}")
            try {
                val clazz = Class.forName(baseBind.etComponent.text.toString())
                if (Activity::class.java.isAssignableFrom(clazz)) {
                    intent.component = ComponentName(this@JumpActivity, clazz)
                    startActivity(intent)
                } else {
                    TipUtil.showToast("${clazz.name} is not a Activity")
                }
            } catch (e: ClassNotFoundException) {
                e.printStackTrace()
                TipUtil.showToast("class not found ${e.message}")
            }
        }
    }

    override fun initData() {}

    override fun onWindowFocusChanged(hasFocus: kotlin.Boolean) {
        super.onWindowFocusChanged(hasFocus)
        baseBind.nsv.updatePadding(top = fixTopHeight, bottom = fixBottomHeight)
    }
}